coderd/database/migrations/000301_system_user.up.sql

```diff
@@ -0,0 +1,51 @@
+ALTER TABLE users
+	ADD COLUMN is_system bool DEFAULT false;
+
+CREATE INDEX user_is_system_idx ON users USING btree (is_system);
+
+COMMENT ON COLUMN users.is_system IS 'Determines if a user is a system user, and therefore cannot login or perform normal actions';
+
+-- TODO: tried using "none" for login type, but the migration produced this error: 'unsafe use of new value "none" of enum type login_type'
+--  	 -> not sure why though? it exists on the login_type enum.
+INSERT INTO users (id, email, username, name, created_at, updated_at, status, rbac_roles, hashed_password, is_system, login_type)
+VALUES ('c42fdf75-3097-471c-8c33-fb52454d81c0', 'prebuilds@system', 'prebuilds', 'Prebuilds Owner', now(), now(),
+		'active', '{}', 'none', true, 'password'::login_type);
+
+-- Create function to check system user modifications
+CREATE OR REPLACE FUNCTION prevent_system_user_changes()
+	RETURNS TRIGGER AS
+$$
+BEGIN
+	IF OLD.is_system = true THEN
+		RAISE EXCEPTION 'Cannot modify or delete system users';
+	END IF;
+	RETURN OLD;
+END;
+$$ LANGUAGE plpgsql;
+
+-- Create trigger to prevent updates to system users
+CREATE TRIGGER prevent_system_user_updates
+	BEFORE UPDATE ON users
+	FOR EACH ROW
+	WHEN (OLD.is_system = true)
+EXECUTE FUNCTION prevent_system_user_changes();
+
+-- Create trigger to prevent deletion of system users
+CREATE TRIGGER prevent_system_user_deletions
+	BEFORE DELETE ON users
+	FOR EACH ROW
+	WHEN (OLD.is_system = true)
+EXECUTE FUNCTION prevent_system_user_changes();
+
+-- TODO: do we *want* to use the default org here? how do we handle multi-org?
```

Comment:
This aspect which will require some follow-up. Prebuilds only works with a single org (the default one) at present. This is because orgs can be added after the fact, and we do not yet have a way in the feature branch (dk/prebuilds) to support multiple orgs.

We will likely need to add a mechanism to the reconciliation loop (see RFC section [here](https://www.notion.so/coderhq/Prebuilds-176d579be59280398851fe6473badfe7?pvs=4#177d579be59280c2a910df77c73ce600)) to detect when prebuilds have been configured on a template which belongs to other orgs, and add the prebuilds user to those orgs.

Is it sensible/feasible to proceed as-is?

Reply:
This, as well as your feedback on the trigger we've added, nudges me toward a step right at the start of the reconciliation loop that:
* checks for the existence of a prebuilds system user
* ensures that it is in the correct state in terms of login type, group membership, etc

That solves the multi-org problem and removes the heavy-handed triggers and function that we've added. Will revert back once I've had a chance to test it.

Reply:
@spikecurtis are you happy for this to be addressed in a subsequent PR?

Reply:
Yeah, reconciliation loop is presumably another PR you're going to post later?

Reply:
https://github.com/coder/internal/issues/526